### PR TITLE
Unit testing

### PR DESCRIPTION
**Pull request by nvnyale**

### CODE REVIEW

File: api/orchestration_movers_test.go [hunk 1]
```diff
@@ -53,8 +54,45 @@ func (d *mockDatasync) CancelTaskExecutionWithContext(ctx context.Context, input
...
+		BytesTransferred:         aws.Int64(100),
+		BytesWritten:             aws.Int64(100),
+		EstimatedBytesToTransfer: aws.Int64(100),
+		EstimatedFilesToTransfer: aws.Int64(100),
+		Excludes:                 []*datasync.FilterRule{},
+		FilesTransferred:         aws.Int64(100),
+		Includes:                 []*datasync.FilterRule{},
+		Options:                  &datasync.Options{},
+		Result:                   &datasync.TaskExecutionResultDetail{},
+		StartTime:                aws.Time(time.Now()),
```
Comment (tenyo) [inline api/orchestration_movers_test.go:86]: This time will never match.
You can define a time variable in this package that you use throughout all the tests, so the time is always the same, e.g.
```
        testTime, err := time.Parse("2006-01-02 15:04:05 -0700 MST", "2020-10-09 23:55:57 +0000 UTC")
	if err != nil {
		t.Errorf("failed to parse time: %s", err)
	}
```
Then here you can do:
```
StartTime: aws.Time(testTime),
```

File: api/orchestration_movers_test.go [hunk 2]
```diff
@@ -203,3 +240,60 @@ func Test_stopTaskRun(t *testing.T) {
 		}
 	}
 }
+
+func TestGetRunListById(t *testing.T) {
+
+	run := &DatamoverRun{
+		BytesTransferred:         aws.Int64(100),
```
Comment (tenyo) [inline api/orchestration_movers_test.go:247]: Use the `testTime` here for `StartTime`

File: api/orchestration_movers_test.go [hunk 3]
```diff
@@ -203,3 +240,60 @@ func Test_stopTaskRun(t *testing.T) {
...
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ctx := context.Background()
+			got, err := o.datamoverRunDescribe(ctx, tt.input.group, tt.input.name, tt.input.id)
+			gotErr := err != nil
+			if tt.exp.isErr != gotErr {
+				t.Error(tt.name, " expected error but did not receive error", err)
+			}
+			if got != nil {
+				got.StartTime = nil
```
Comment (tenyo) [inline api/orchestration_movers_test.go:294]: and you should be able to get rid of this

File: api/orchestration_movers_test.go [hunk 4]
```diff
@@ -203,3 +240,60 @@ func Test_stopTaskRun(t *testing.T) {
...
+		isErr bool
+		run   *DatamoverRun
+	}
+	tests := []struct {
+		name  string
+		input input
+		exp   output
+	}{
+		{"group empty", input{"", "name1", "1234"}, output{true, nil}},
+		{"name empty", input{"group1", "", "1234"}, output{true, nil}},
```
Comment (tenyo) [inline api/orchestration_movers_test.go:271]: add another test for empty `id`

File: api/orchestration_movers_test.go [hunk 5]
```diff
@@ -53,8 +54,45 @@ func (d *mockDatasync) CancelTaskExecutionWithContext(ctx context.Context, input
 
 var is_running = false
 
+func (d *mockDataSync) ListTaskExecutionsPagesWithContext(ctx context.Context, input *datasync.ListTaskExecutionsInput, callback func(*datasync.ListTaskExecutionsOutput, bool) bool, opts ...request.Option) error {
```
Comment (tenyo) [inline api/orchestration_movers_test.go:57]: Were you also planning to test `datamoverRunList` below? I only see `datamoverRunDescribe`

**Review by tenyo — COMMENTED** [review]
Thanks, I left some comments/questions.
I think using the assert is OK if it's easier, although you can probably accomplish the same thing with `reflect.DeepEqual`

File: api/orchestration_movers_test.go (hunk 1 above)
Comment (nvnyale) [reply api/orchestration_movers_test.go]: Done..

File: api/orchestration_movers_test.go (hunk 2 above)
Comment (nvnyale) [reply api/orchestration_movers_test.go:236]: Done.

File: api/orchestration_movers_test.go (hunk 3 above)
Comment (nvnyale) [reply api/orchestration_movers_test.go]: Done..

File: api/orchestration_movers_test.go (hunk 4 above)
Comment (nvnyale) [reply api/orchestration_movers_test.go:261]: added..done.

File: api/orchestration_movers_test.go (hunk 5 above)
Comment (nvnyale) [reply api/orchestration_movers_test.go:56]: Added Datarunlist.

File: api/orchestration_movers_test.go [hunk 6]
```diff
@@ -203,3 +229,121 @@ func Test_stopTaskRun(t *testing.T) {
...
+				t.Error("expected error but did not receive error")
+			} else if !tt.exp.isErr && err != nil {
+				t.Errorf("received unexpected error %v", err)
+			}
+			assert.Equal(t, tt.exp.res, got)
+		})
+	}
+}
+
+func TestGetMoversList(t *testing.T) {
```
Comment (nvnyale) [inline api/orchestration_movers_test.go:318]: All fields of the datasyncorchestrator are concrete structs, so we won't be able test all the code paths properly. Replacing those fields with interfaces will allow us to mock them out and properly test all the code paths in orchestrator_movers.go

File: api/orchestration_movers_test.go [hunk 7]
```diff
@@ -203,3 +229,121 @@ func Test_stopTaskRun(t *testing.T) {
 		}
 	}
 }
+
+func TestGetRunListById(t *testing.T) {
```
Comment (tenyo) [inline api/orchestration_movers_test.go:233]: Let's rename this to `Test_datamoverRunDescribe` since that's the func we're testing

File: api/orchestration_movers_test.go [hunk 8]
```diff
@@ -203,3 +229,121 @@ func Test_stopTaskRun(t *testing.T) {
...
+				t.Error("expected error but did not receive error")
+			} else if !tt.exp.isErr && err != nil {
+				t.Errorf("received unexpected error %v", err)
+			}
+			assert.Equal(t, tt.exp.run, got)
+		})
+	}
+}
+
+func TestGetRunList(t *testing.T) {
```
Comment (tenyo) [inline api/orchestration_movers_test.go:282]: `Test_datamoverRunList`

File: api/orchestration_movers_test.go (hunk 6 above)
Comment (tenyo) [inline api/orchestration_movers_test.go:318]: `Test_datamoverList`

Comment (tenyo) [reply api/orchestration_movers_test.go:318]: I'm not sure what you mean exactly, but I don't think it's a good idea to use interfaces (unless absolutely necessary).

**Review by tenyo — COMMENTED** [review]
Just a few minor things otherwise I think it's good 👍

File: api/orchestration_movers_test.go (hunk 7 above)
Comment (nvnyale) [reply api/orchestration_movers_test.go]: Renamed.

File: api/orchestration_movers_test.go (hunk 8 above)
Comment (nvnyale) [reply api/orchestration_movers_test.go]: Renamed.

File: api/orchestration_movers_test.go (hunk 6 above)
Comment (nvnyale) [reply api/orchestration_movers_test.go]: Renamed.

**Review by tenyo — APPROVED** [review]
